sdk/include/opentelemetry/sdk/common/global_log_handler.h

```diff
@@ -136,74 +147,71 @@ class GlobalLogHandler
 
-#define OTEL_INTERNAL_LOG_DISPATCH(level, message, attributes)                                     \
-  do                                                                                               \
-  {                                                                                                \
-    using namespace opentelemetry::sdk::common::internal_log;                                      \
-    std::stringstream tmp_stream;                                                                  \
-    tmp_stream << message;                                                                         \
-    GlobalLogHandler::GetLogHandler()->Handle(level, __FILE__, __LINE__, tmp_stream.str().c_str(), \
-                                              attributes);                                         \
-  } while (0)
+/**
+ * We can not decide the destroying order of signaltons.
+ * Which means, the destructors of other singletons (GlobalLogHandler,TracerProvider and etc.)
+ * may be called after destroying of global LogHandler and use OTEL_INTERNAL_LOG_* in it.We can do
+ * nothing but ignore the log in this situation.
+ */
+#define OTEL_INTERNAL_LOG_DISPATCH(level, message, attributes)                            \
+  do                                                                                      \
+  {                                                                                       \
+    using opentelemetry::sdk::common::internal_log::GlobalLogHandler;                     \
+    using opentelemetry::sdk::common::internal_log::LogHandler;                           \
+    if (level > GlobalLogHandler::GetLogLevel())                                          \
+    {                                                                                     \
+      break;                                                                              \
+    }                                                                                     \
+    const opentelemetry::nostd::shared_ptr<LogHandler> &log_handler =                     \
+        GlobalLogHandler::GetLogHandler();                                                \
+    if (!log_handler)                                                                     \
+    {                                                                                     \
+      break;                                                                              \
+    }                                                                                     \
+    std::stringstream tmp_stream;                                                         \
+    tmp_stream << message;                                                                \
+    log_handler->Handle(level, __FILE__, __LINE__, tmp_stream.str().c_str(), attributes); \
+  } while (false);
 
 #define OTEL_INTERNAL_LOG_GET_3RD_ARG(arg1, arg2, arg3, ...) arg3
 
-#if OTEL_INTERNAL_LOG_LEVEL >= OTEL_INTERNAL_LOG_LEVEL_ERROR
```

Comment:
Do you think we should keep OTEL_INTERNAL_LOG_LEVEL along with the run-time level? If this macro is defined, it should override the run-time levels ( it setting run-time level will not have any effect). One of the advantages of having a compile-time macro OTEL_INTERNAL_LOG_LEVEL is that macro-call `OTEL_INTERNAL_LOG_DISPATCH` could be totally removed from the hot-path of the event pipeline based on the level defined. We can let it not be defined by default in our build, but users have the option to enable it if they want.

Reply:
Yes, of course. Can we define the use `OTEL_INTERNAL_LOG_LEVEL_DEBUG` as the default value of `OTEL_INTERNAL_LOG_LEVEL` ? Many users use the prebuilt packages provided by package manager and need debug log to test integration.

Reply:
Yes, that would be good too. Thanks.